lib/rules/disallow-identifier-names.js

```diff
@@ -54,14 +54,14 @@ module.exports.prototype = {
         var disallowedIdentifiers = this._identifierIndex;
 
         file.iterateNodesByType('Identifier', function(node) {
-            if (disallowedIdentifiers[node.name]) {
+            if (disallowedIdentifiers.hasOwnProperty(node.name)) {
```

Comment:
Actually this will cause a problem if a `hasOwnProperty` key exists on the object. You want `Object.prototype.hasOwnProperty.call(disallowedIdentifiers, node.name)`